frontend/src/utils/filterHelpers.js

```diff
@@ -322,3 +328,44 @@ export function parseFiltersToPatch(params) {
     return acc;
   }, []);
 }
+
+const cleanupParameter = (param) => {
+  const { parameterName, defaultValue, defaultValueTo, widgetType } = param;
+  let { value, valueTo } = param;
+
+  if (widgetType === 'Date' && value) {
+    value = Moment(value).format(DATE_FORMAT);
+  }
+  if (widgetType === 'Date' && valueTo) {
+    valueTo = Moment(valueTo).format(DATE_FORMAT);
+  }
+
+  value = value === null && defaultValue ? defaultValue : value;
```

Comment:
this was the actual issue of this task.

pls add a comment explaining why we are doing this
asking because, correct me if I am wrong, it's not intuitive at all.